### PR TITLE
Generate a Kotlin wrapper with Kotlin Poet

### PR DESCRIPTION
**Pull request by jmfayard**
I am able to generate the file `SetupnodeV2.kt` with KotlinPoet

### CODE REVIEW

File: wrapper-generator/src/main/kotlin/KotlinPoet.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        )
+        // @file:Suppress("RedundantVisibilityModifier")
+        .addAnnotation(AnnotationSpec.builder(Suppress::class).addMember("%S", "RedundantVisibilityModifier").build())
+        .build()
+
+fun Manifest.toYamlFunction(): FunSpec {
+    return FunSpec.builder("toYamlArguments")
+        .returns(LinkedHashMap::class.parameterizedBy(String::class, String::class))
+        .addModifiers(KModifier.OVERRIDE)
+        .addCode("return TODO(%S)", "You need to implement toYamlArguments()")
```
Comment (krzema12) [inline wrapper-generator/src/main/kotlin/KotlinPoet.kt:60]: This is a must-have to have MVP :)

File: wrapper-generator/src/main/kotlin/Generation.kt [hunk 2]
```diff
@@ -1,5 +1,35 @@
-fun ActionCoords.generateWrapper(): String {
-    val manifest = fetchManifest()
-    println(manifest)
-    return ""
+data class Wrapper(
+    val kotlinCode: String,
+    val filePath: String,
+)
+
+fun ActionCoords.generateWrapper(
```
Comment (krzema12) [inline wrapper-generator/src/main/kotlin/Generation.kt:6]: If we're going the kotlinpoet way, this function is not needed right?

File: wrapper-generator/src/test/kotlin/GenerationTest.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+
+class GenerationTest : FunSpec({
```
Comment (krzema12) [inline wrapper-generator/src/test/kotlin/GenerationTest.kt:6]: Let's have some tests for the kotlinpoet approach.

**Review by krzema12 — CHANGES_REQUESTED** [review]
Great direction, I like it :)
To make a consistent change, let's make the SetupNodeV2 wrapper fully functional.

File: wrapper-generator/src/main/kotlin/Generation.kt (hunk 2 above)
Comment (jmfayard) [reply wrapper-generator/src/main/kotlin/Generation.kt]: Removed

File: wrapper-generator/src/main/kotlin/KotlinPoet.kt (hunk 1 above)
Comment (jmfayard) [reply wrapper-generator/src/main/kotlin/KotlinPoet.kt]: Done

File: wrapper-generator/src/test/kotlin/GenerationTest.kt (hunk 3 above)
Comment (jmfayard) [reply wrapper-generator/src/test/kotlin/GenerationTest.kt]: Done

**Comment by jmfayard** [comment]
@krzema12 
- [The Wrapper now supporst Parameters of type boolean and integer](https://github.com/krzema12/github-actions-kotlin-dsl/pull/75/commits/df6ec8e3dccd33612cc09343e627283215cea874) - based on the default value of the parameter
- the function `toYamlArguments()` is now automatically generated
- the test class MyActionTest is now automatically generated

**Comment by jmfayard** [comment]
@krzema12 I added a command to generate the wrapper from the command-line

`./gradlew :wrapper-generator:run --args https://github.com/peterjgrainger/action-create-branch/releases/tag/v2.1.0`

I'm now done feature-wise

**Comment by krzema12** [comment]
Please don't bother about the merge conflicts, I'm introducing these changes in smaller chunks and with slightly different approach, using you PR as inspiration/reference implementation. I'll mark you as author wherever necessary. Thanks for your effort! :tada:

**Pull request closed without merging**